modules/Bio/EnsEMBL/Production/Pipeline/Common/CheckAssemblyGeneset.pm

```diff
@@ -0,0 +1,91 @@
+
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+ Bio::EnsEMBL::Production::Pipeline::Common::CheckAssemblyGeneset;
+
+=head1 DESCRIPTION
+
+ Module to check if a species has an updated assembly or geneset for a given release
+ The module uses the metadata database and compare given release with previous release
+
+=cut
+
+package Bio::EnsEMBL::Production::Pipeline::Common::CheckAssemblyGeneset;
+
+use strict;
+use warnings;
+use Bio::EnsEMBL::MetaData::DBSQL::MetaDataDBAdaptor;
+use Bio::EnsEMBL::MetaData::Base qw(process_division_names fetch_and_set_release check_assembly_update check_genebuild_update);
+use base qw/Bio::EnsEMBL::Production::Pipeline::Common::Base/;
+
+sub write_output {
+  my ($self) = @_;
+  my $release = $self->param_required('release');
+  my $species = $self->param_required('species');
+  my $run_all = $self->param('run_all');
```

Comment:
I don't think we should be re-using the parameter name 'run_all'; that is used with Db/SpeciesFactory modules, and has a different meaning here. Suggest changing to 'always_run'.
Also, this parameter value could be tested right at the top of the method; if it's set, we don't need to bother attaching to the metadata db to try to find a previous version.

Reply:
My idea was to reuse this flag but your are right it might not be appropriate. I've added a new flag called "skip_metadata_check" working as you describe above.